scipy-1.0/paper.tex

```diff
@@ -298,25 +298,26 @@ \subsubsection*{LowLevelCallable}
 possible to generate a low-level callback function automatically
 from a Cython module using \texttt{scipy.LowLevelCallable.from\_cython}.
 
-\subsection*{Cython bindings for BLAS, LAPACK and special}
-
-When writing algorithms or wrappers in Cython\cite{behnel2011cython}, it is often convenient to call directly into the libraries that SciPy wraps rather than routing the call back through the Python interpreter.
-SciPy currently includes Cython wrappers\cite{blas-lapack-wrappers-scipy-2015} for many BLAS and LAPACK\cite{LAPACK} routines and for many of the special functions provided in the \texttt{scipy.{\allowbreak}special} submodule.
-These wrappers are available in the modules \texttt{scipy.{\allowbreak}linalg.{\allowbreak}cython\_blas}, \texttt{scipy.{\allowbreak}linalg.{\allowbreak}cython\_lapack}, and \texttt{scipy.{\allowbreak}special.{\allowbreak}cython\_special} respectively.
-The low-level interfaces for Cython can also be used by third party libraries to get direct access to the functions in these libraries while avoiding the overhead of Python function calls{\textemdash}this can give performance gains of one or two orders of magnitude for many use cases.
-Cython's cimport machinery allows Cython based modules to share C-level objects with other Cython modules without requiring that the third-party extensions link against any additional libraries.
-Function pointers to wrappers for library routines can also be shared in this way, so that Cython modules using the interface provided by SciPy no longer have to worry about finding and linking against the original library.
-This lets other extensions avoid the complexity of finding and using the correct libraries on the various possible Fortran compiler ABI and OS combinations.
-The low-level Cython wrappers provide an OS and Fortran compiler independent interface for the libraries they wrap.
-
-To help with both correctness and ease of maintenance, most of these low-level Cython wrappers are generated automatically.
+\subsection*{Cython bindings for BLAS, LAPACK, and special}
+
+SciPy includes Cython\cite{behnel2011cython} wrappers for many BLAS and LAPACK\cite{LAPACK} routines and for the special functions provided in the \texttt{scipy.{\allowbreak}special} submodule.
```

Comment:
Is Cython explained elsewhere in the paper?  Otherwise, we need an introduction that tells the reader what Cython in, why it is being used (not three paragraphs down, as is currently the case).  Something along the lines of:

Cython is a Python to C compiler, utilized by SciPy to gain speed improvements and to wrap low-level C and C++ libraries.  For example, ... [ckdtrees, special, blas] ...

The sentence later "When writing algorithms in Cython, it is typically more efficient..." can then be removed.

Reply:
@insertinterestingnamehere that suggestion makes sense to me

Reply:
It can introduce Cython in the intro/history section.  I was already planning to mention it, so it would make sense to explain it as well.

Reply:
That seems like a good idea. Perhaps we can introduce it there and then elaborate a little more on why we use it in an independent paragraph in the language choices section?

Reply:
Sounds good. Seems like there aren't any changes needed to this section then.